allure-testng/build.gradle.kts

```diff
@@ -2,7 +2,7 @@ description = "Allure TestNG Integration"
 
 val agent: Configuration by configurations.creating
 
-val testNgVersion = "6.14.3"
```

Comment:
The question is -- should we drop support of TestNG 6 (according to the statistics it still have much more users than TestNG 7)? The other option would be create separate allure-testng7 integration

Reply:
Hm, good question... I would say it's up to You :) From my point of view 6.X branch is so old, that it's the highest time to move on. They don't even consider any issues affecting to 6.X branch. 
What's more - I don't see too many public API changes in 7.X. Even in the allure-testng tests suite, only 1 test was slightly affected:
https://github.com/allure-framework/allure-java/pull/503/files#diff-092fdd6ce32f9699de58adbcc839cbec2fc50c076a32404a1592154ae42e7fb3R437. 
 I think both ways would do the job here.

Reply:
There were made some important updates in interfaces in 7+ versions. So be aware of the fact, that merging this PR may break existing users' code and force them to migrate to a new TestNG version.

Reply:
I think that all the methods in the interfaces were made default, including new interfaces if any. So, all the changes in the interfaces should be transparent for a user. The only requirement would be to migrate to J8. But if one is not on J8 already, then I don't think he/she would consider bumping the allure version :) 
As @baev proposed, You might want to consider kicking off a new TNG7 project though. Anyway, I would recommend considering slowly deprecating the 6.X branch.

Reply:
I perfectly know about the default interfaces, as it was my PR. :)

I was talking about a bit different stuff, e.g. there was a mess of annotation transformers (1,2,3) in 6.x versions, which have been reduced to a single instance since 7+.

So if people use an already removed from 7.x version transformers, updating Allure will introduce a transitive dependency which will break their code. And they won't understand why.

Personally, I'm ok to update TestNG dependency, but there should be a bolded release note about potential breaking changes.